src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -189,20 +239,32 @@ private IServiceCollection BuildCommonServices()
             var services = new ServiceCollection();
             services.AddSingleton(_hostingEnvironment);
 
+            var builder = new ConfigurationBuilder();
+            if(_configureConfigurationBuilderDelegate != null)
+            {
+                _configureConfigurationBuilderDelegate(builder, _hostingEnvironment);
+            }
+            services.AddSingleton<IConfiguration>(builder.Build());
+
+            ILoggerFactory loggerFactory;
             // The configured ILoggerFactory is added as a singleton here. AddLogging below will not add an additional one.
-            if (_loggerFactory == null)
+            if (_loggerFactoryFunc == null)
             {
-                _loggerFactory = new LoggerFactory();
-                services.AddSingleton(provider => _loggerFactory);
+                loggerFactory = new LoggerFactory();
             }
             else
             {
-                services.AddSingleton(_loggerFactory);
+                loggerFactory = _loggerFactoryFunc(_hostingEnvironment, _config);
```

Comment:
null check on user results? Or collapse the whole thing to:
`loggerFactory = _loggerFactoryFunc?(_hostingEnvironment, _config) ?? new LoggerFactory();`